### PR TITLE
updated communication_interfaces folder

### PR DESCRIPTION
**Pull request by haripriya9647**
## Description
added inline docs for methods and class along with examples

## Relevant Issue
Issue #269  the issue here

## Affected Dependencies
None
## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/NimbleEdge/EnvisEdge/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/NimbleEdge/EnvisEdge/blob/main/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [NimbleEdge Styleguide](https://numpydoc.readthedocs.io/en/latest/format.html)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/NimbleEdge/EnvisEdge/labels)
- [x] My changes are covered by tests

### CODE REVIEW

File: fedrec/communication_interfaces/abstract_comm_manager.py [hunk 1]
```diff
@@ -9,6 +9,14 @@
 
 
 class AbstractCommunicationManager(ABC):
+    """
+    This class works effectively on commuting messages between workers and job executors , and also performs different message operations like sending, receiving,serializing and deserializing.
+
+    Serialization is the process of converting a data object into a format that can be reused later. It helps in reconstructing the object later and helps in obtaining the exact structure/object which makes it easy to use instead of reconstructing the object from scratch.
+
+    Deserialization is the reverse process of serialization. It reconstructs the data object from a set of formats.
```
Comment (divyanshugit) [inline fedrec/communication_interfaces/abstract_comm_manager.py:17]: Re-write it according to pep8(79 letters in a single line).

File: fedrec/communication_interfaces/zeroMQ_interface.py [hunk 2]
```diff
@@ -6,6 +6,33 @@
 
 @registry.load("communication_interface", "ZeroMQ")
 class ZeroMQ(AbstractCommunicationManager):
+    """
+    ZeroMQ class implements the basic send/receive interface between the publisher and the subscriber. Senders of messages token are called publishers and the one who receives these tokens are called subscribers.
+
+
+    It is a high-performance messaging library, which is aimed to use in distributed and concurrent environments.The philosophy of ZeroMQ starts with zero where the zero is for zero broker, zero latency, zero cost, and zero administration.
+
```
Comment (divyanshugit) [inline fedrec/communication_interfaces/zeroMQ_interface.py:14]: Re-write it and get rid of unnecessary blank lines.

**Review by divyanshugit — CHANGES_REQUESTED** [review]

**Comment by divyanshugit** [comment]
Please add a proper description of the PR and an example as well for `zeroMQ_interface.py`

File: fedrec/communication_interfaces/zeroMQ_interface.py [hunk 3]
```diff
@@ -19,6 +19,45 @@ class ZeroMQ(AbstractCommunicationManager):
     for zero broker, zero latency, zero cost, and zero
```
Comment (ramesht007) [inline fedrec/communication_interfaces/zeroMQ_interface.py:19]: We can remove this paragraph.

**Review by ramesht007 — CHANGES_REQUESTED** [review]
LGTM

File: fedrec/communication_interfaces/zeroMQ_interface.py (hunk 3 above)
Comment (haripriya9647) [reply fedrec/communication_interfaces/zeroMQ_interface.py:19]: @ramesht007 are you referring to the whole paragraph or just the second para??

**Review by ramesht007 — APPROVED** [review]
LGTM!!

**Pull request merged**